src/plugins/vis_builder/public/application/components/data_tab/index.tsx

```diff
@@ -3,19 +3,336 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { DropResult, EuiDragDropContext } from '@elastic/eui';
 import { FieldSelector } from './field_selector';
 
 import './index.scss';
 import { ConfigPanel } from './config_panel';
+import { useAggs, useVisualizationType } from '../../utils/use';
+import { useTypedDispatch, useTypedSelector } from '../../utils/state_management';
+
+import {
+  reorderAgg,
+  updateAggConfigParams,
+} from '../../utils/state_management/visualization_slice';
+import { BucketAggType, IndexPatternField, propFilter } from '../../../../../data/common';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+import { VisBuilderServices } from '../../../types';
 
 export const DATA_TAB_ID = 'data_tab';
 
+const getIndexPatternField = (indexFieldName: string, availableFields: IndexPatternField[]) =>
+  availableFields.find(({ name }) => name === indexFieldName);
+
+const filterByName = propFilter('name');
+const filterByType = propFilter('type');
+
+export const getValidAggTypes = ({
```

Comment:
This one is similar to another function here

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/5ee6f589ab6f25ed97a8fc4f7f3139b3b30631eb/src/plugins/vis_builder/public/application/components/data_tab/use/use_dropbox.tsx#L159

The two functions getValidAggTypes and getValidAggTypes (with parameters) are very similar. Both functions aim to determine the valid aggregation types based on the schema and the field being dragged or manipulated. They follow a similar logic flow to achieve this.

I would suggest that we consolidate the two functions into a single function that can handle both scenarios by using a more flexible parameter structure.

Maybe move this one, getIndexPatternField and createNewAggConfig to a util file if they are repeated used?

Reply:
I have consolidated the common functions into a util and reused it across required components.